test/e2e/setup/rosa/setup/maestro.sh

```
@@ -56,81 +58,213 @@ maestro_cert_arn=$(aws iot create-keys-and-certificate \
-subnets_counts=0
-for subnet in $(aws ec2 describe-subnets --region ${region} --filters "Name=vpc-id,Values=${vpc}" | jq -r '.Subnets[].SubnetId'); do
-    subnets="$subnets,\"$subnet\""
-    subnets_counts=$((subnets_counts+1))
+declare -a unique_azs
+for subnet_data in $(aws ec2 describe-subnets --region ${region} --filters "Name=vpc-id,Values=${vpc}" --output json | jq -c '.Subnets[] | {id: .SubnetId, az: .AvailabilityZone}'); do
+    subnet_id=$(echo "$subnet_data" | jq -r '.id')
+    subnet_az=$(echo "$subnet_data" | jq -r '.az')
+    subnets="$subnets,\"$subnet_id\""
+
+    # Track unique AZs
+    if [[ ! " ${unique_azs[@]} " =~ " ${subnet_az} " ]]; then
```

Shellcheck (0.11.0)

[error] Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

(SC2199)